src/main/kotlin/no/nav/syfo/melding/domain/MeldingTilBehandler.kt

```diff
@@ -84,28 +84,45 @@ private fun MeldingTilBehandler.getBrevKode(): BrevkodeType {
     }
 }
 
-fun MeldingTilBehandler.toJournalpostRequest(pdf: ByteArray) = JournalpostRequest(
-    avsenderMottaker = createAvsenderMottaker(behandlerPersonIdent, behandlerNavn),
-    tittel = "Dialogmelding til behandler",
-    bruker = Bruker.create(
-        id = arbeidstakerPersonIdent.value,
-        idType = BrukerIdType.PERSON_IDENT,
-    ),
-    dokumenter = listOf(
-        Dokument.create(
-            brevkode = this.getBrevKode(),
-            tittel = "Dialogmelding til behandler",
-            dokumentvarianter = listOf(
-                Dokumentvariant.create(
-                    filnavn = "Dialogmelding til behandler",
-                    filtype = FiltypeType.PDFA,
-                    fysiskDokument = pdf,
-                    variantformat = VariantformatType.ARKIV,
-                )
-            ),
-        )
-    ),
-)
+fun MeldingTilBehandler.toJournalpostRequest(pdf: ByteArray, isPaminnelse: Boolean) =
+    JournalpostRequest(
+        avsenderMottaker = createAvsenderMottaker(behandlerPersonIdent, behandlerNavn),
+        tittel = createTittel(isPaminnelse),
+        bruker = Bruker.create(
+            id = arbeidstakerPersonIdent.value,
+            idType = BrukerIdType.PERSON_IDENT,
+        ),
+        dokumenter = listOf(
+            Dokument.create(
+                brevkode = this.getBrevKode(),
+                tittel = createTittel(isPaminnelse),
+                dokumentvarianter = listOf(
+                    Dokumentvariant.create(
+                        filnavn = createTittel(isPaminnelse),
+                        filtype = FiltypeType.PDFA,
+                        fysiskDokument = pdf,
+                        variantformat = VariantformatType.ARKIV,
+                    )
+                ),
+            )
+        ),
+        overstyrInnsynsregler = createOverstyrInnsynsregler(!isPaminnelse),
+    )
+
+fun createTittel(isPaminnelse: Boolean): String {
```

Comment:
Foreslår at `createTittel` og `createOverstyrInnsynsregler` er extension functions på `MeldingTilBehandler` og bruker `when (this.type)` på samme måte som `fun MeldingTilBehandler.getBrevKode()` f.eks. Da slipper man å bruke boolsk flagg isPaminnelse osv fra cronjobben og det blir litt mer robust ved utvidelse med nye meldingstyper.

Reply:
Ser på det, takk for forslag 👍🏻